src/AppInstallerCommonCore/Certificates.cpp

```diff
@@ -0,0 +1,511 @@
+            std::wstring result(characterCount, L'\0');
+            characterCount = CertGetNameStringW(certContext, CERT_NAME_SIMPLE_DISPLAY_TYPE, 0, nullptr, &result[0], characterCount);
+
+            if (static_cast<size_t>(characterCount) == result.size())
+            {
+                return Utility::ConvertToUTF8(static_cast<std::wstring_view>(result).substr(0, result.size() - 1));
+            }
+            else
+            {
+                return "<unknown>";
+            }
+        }
+
+        std::string GetDescriptionOfCertChain(PCCERT_CHAIN_CONTEXT chainContext)
+        {
+            PCCERT_SIMPLE_CHAIN chain = chainContext->rgpChain[0];
+            std::ostringstream stream;
+            std::string indent;
+
+            for (DWORD i = 0; i < chain->cElement; ++i)
+            {
+                PCCERT_CHAIN_ELEMENT element = chain->rgpElement[(chain->cElement - 1) - i];
+
+                if (!indent.empty())
+                {
+                    stream << std::endl;
+                }
+
+                stream << indent;
+
+                stream << GetSimpleDisplayName(element->pCertContext);
+
+                indent.append("  ");
+            }
+
+            return std::move(stream).str();
+        }
+
+        std::optional<PinningVerificationType> GetTypeFromString(std::string_view value)
+        {
+            std::string lowerValue = Utility::ToLower(value);
+
+            if (lowerValue == "none")
+            {
+                return PinningVerificationType::None;
+            }
+            else if (lowerValue == "publickey")
+            {
+                return PinningVerificationType::PublicKey;
+            }
+            else if (lowerValue == "subject")
+            {
+                return PinningVerificationType::Subject;
+            }
+            else if (lowerValue == "issuer")
+            {
+                return PinningVerificationType::Issuer;
+            }
+
+            return {};
+        }
+    }
+
+    PinningDetails& PinningDetails::LoadCertificate(int resource)
+    {
+        return LoadCertificate(Resource::GetResourceAsBytes(resource, CERTIFICATE_RESOURCE_TYPE));
+    }
+
+    PinningDetails& PinningDetails::LoadCertificate(const std::vector<BYTE>& certificateBytes)
+    {
+        return LoadCertificate(std::make_pair(&certificateBytes[0], certificateBytes.size()));
+    }
+
+    PinningDetails& PinningDetails::LoadCertificate(const std::pair<const BYTE*, size_t> certificateBytes)
+    {
+        m_certificateContext.reset(CertCreateCertificateContext(X509_ASN_ENCODING | PKCS_7_ASN_ENCODING, certificateBytes.first, static_cast<DWORD>(certificateBytes.second)));
+        THROW_LAST_ERROR_IF(!m_certificateContext);
+        return *this;
+    }
+
+    PinningDetails& PinningDetails::SetPinning(PinningVerificationType type)
+    {
+        m_pinning = type;
+        return *this;
+    }
+
+    bool PinningDetails::LoadFrom(const Json::Value& configuration)

```

Comment:
>PinningDetails::LoadFrom [](http://example.com/codeflow?start=9&length=24)

A comment with sample configuration json would better help with reading the parsing logic